include/fastpath/fastpath.h

```diff
@@ -114,5 +119,96 @@ static inline reply_t *thread_state_get_replyObject_np(thread_state_t ts)
 }
 #endif
 
+static inline void cancelSignal_fp(tcb_t *tptr, notification_t *ntfnPtr)
```

Comment:
how different are these to the existing cancel functions? Is it really worth the duplication?

Reply:
Not very different from what I understand. @oliver-wm suggested that I implement them [here](https://github.com/seL4/seL4/pull/322#discussion_r602050907).

Reply:
i recommend inlining the functions and adding `likely` to any branches. This will favour the fastpath. 

The only case for duplication is when you want to select a subset of the behaviour from a function that is too big to inline. 

perhaps I should write a 'so  you want to add a fastpath' guide :)